### PR TITLE
👷‍♂️ Add wheel to Azure Artifacts

### PR DESCRIPTION
**Pull request by edjubuh**
#### Summary:
Adds CLI wheel to the Azure Artifacts repository.

#### Motivation:
We need to use the CLI in other pipelines (like docs and kernel build) and this is a convenient way to accomplish that

##### References (optional):
purduesigbots/pros-docs#100

#### Test Plan:
- [X] It build and uploads

### CODE REVIEW

**Comment by edjubuh** [comment]
So it's hard for me to test the conditions without actually merging into `develop` first, but I'm pretty sure it's right. Everything else is correct though, as I tested without the condition to make sure it would actually publish the wheel.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -71,5 +75,22 @@ jobs:
         artifactName: 'pros_cli-$(Build.BuildNumber)-linux-$(buildArch)'
         targetPath: 'out'
 
-
+- job: PublishArtifact
```
Comment (HotelCalifornia) [inline azure-pipelines.yml:78]: Is this manually doing the work of `PublishPipelineArtifact@0`? Unless I'm mistaken and that task just makes artifacts available within the same pipeline... (Still though, I'd have though there would be a built-in task to do this kind of injection)

Comment (edjubuh) [reply azure-pipelines.yml:78]: No, this is just publishing to Azure Artifacts, which is different.

Comment (HotelCalifornia) [reply azure-pipelines.yml:78]: Alright that's fine then. Still feel like they should, like, provide a task that does this for us

Comment (edjubuh) [reply azure-pipelines.yml:78]: [¯\\_(ツ)_/¯](https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pypi)

File: azure-pipelines.yml [hunk 2]
```diff
@@ -71,5 +75,22 @@ jobs:
...
+  steps:
+    - checkout: none
+    - task: DownloadPipelineArtifact@0
+      inputs:
+        artifactName: 'pros_cli-$(Build.BuildNumber)-linux-x64'
+        targetPath: $(System.DefaultWorkingDirectory)
+    - script: 'sudo pip install twine'
+    - task: TwineAuthenticate@0
+      inputs:
+        artifactFeeds: 'pros'
```
Comment (HotelCalifornia) [inline azure-pipelines.yml]: Can you rename this feed to `pros-cli`? It's a little more explicit and we may want to do something with the kernel on azure in the future

**Review by HotelCalifornia — APPROVED** [review]

**Pull request merged**